ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/operations/AttestationData.java

```diff
@@ -70,15 +70,19 @@ public AttestationData(UInt64 slot, AttestationData data) {
   }
 
   public UInt64 getEarliestSlotForForkChoice(final Spec spec) {
-    // Attestations can't be processed by fork choice until their slot is in the past and until we
-    // are in the same epoch as their target.
-    return getSlot().plus(UInt64.ONE).max(getTarget().getEpochStartSlot(spec));
+    final UInt64 targetEpochStartSlot = getTarget().getEpochStartSlot(spec);
+    return getEarliestSlotForForkChoice(targetEpochStartSlot);
   }
 
   public UInt64 getEarliestSlotForForkChoice(final MiscHelpers miscHelpers) {
-    return getSlot()
-        .plus(UInt64.ONE)
-        .max(miscHelpers.computeStartSlotAtEpoch(getTarget().getEpoch()));
+    final UInt64 targetEpochStartSlot = miscHelpers.computeStartSlotAtEpoch(getTarget().getEpoch());
+    return getEarliestSlotForForkChoice(targetEpochStartSlot);
+  }
+
+  public UInt64 getEarliestSlotForForkChoice(final UInt64 targetEpochStartSlot) {
+    // Attestations can't be processed by fork choice until their slot is in the past and until we
+    // are in the same epoch as their target.
+    return getSlot().plus(1).max(targetEpochStartSlot);
```

Comment:
should use the constant here for less conversions (UInt64.ONE) like initial implementation.

Reply:
For some reason in my head thought using a primitive uses less conversion. Will fix tomorrow.